### ISSUE TITLE
[Discussion] Don't sort completion list when it's incomplete and there's no sort text set.

### ISSUE THREAD

**Issue by kiennq** [emacs-lsp/lsp-mode#1661]
Basically I think we should probably use the logic in [this comment](https://github.com/microsoft/language-server-protocol/issues/898#issuecomment-582120134), with a few changes:
- if `filterText` is null, the element must always be shown and never filtered by the client (already done)
- if completion list is incomplete and all of the `sortText` is not set, the client mustn't sort the list.

The second one are quite reasonable given that the `sortText` are optional and the completion list is an array and have order preserved via json encoding. The requirement of incomplete completion list is obvious, once the completion list is complete, it will be cached by client and the client side filtering + re-sorting will be needed.

WDYT? @emacs-lsp/lsp-mode @yyoncho  @tpadioleau

**Comment by yyoncho** [comment]
> if completion list is incomplete and all of the sortText is not set, the client mustn't sort the list.

I think it makes sense.

Also, IMO it will be good to document what are the corner cases and what are the servers that we handle with a particular strategy. E. g. prefix mismatch handling <describe our strategy> - servers that have that ts-ls/jdt-ls.

**Comment by tpadioleau** [comment]
Logic seems fine to me.

**Comment by kiennq** [comment]
I think we can close this one now.
